addons/l10n_fr/models/res_partner.py

```diff
@@ -7,3 +7,19 @@ class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     siret = fields.Char(string='SIRET', size=14)
+
+    def _get_peppol_country_code(self):
+        country_code = super()._get_peppol_country_code()
+        if self.siret:
+            return 'FR'
+        return country_code
+
+    @api.depends('siret')
```

Comment:
for all those dependencies, maybe you can try something more sophisticated: you can add a lambda in a `depends` :smirk: 
Example:

```py
    @api.depends(lambda self: self._get_fields_breaking_publication())
    def _compute_publication_warning(self):
        for slot in self:
            slot.publication_warning = slot.resource_id and slot.resource_type != 'material' and slot.state == 'published'

```

Reply:
:exploding_head: